tests/test_einsum.py

```diff
@@ -53,6 +174,47 @@ def test_all():
                 output = backend.to_numpy(output_framework)
                 assert output.shape == out_shape
 
+
+def test_functional():
+    # Functional tests:
+    valid_backends = ['tensorflow', 'torch', 'jax', 'numpy',
+                      'chainer', 'oneflow', 'cupy', 'tensorflow.keras']
+    for backend in collect_test_backends():
```

Comment:
important - this collects only non-symbolic frameworks.
Need one more similar section with symbolic frameworks, the only difference is computation part itself, see here:

https://github.com/arogozhnikov/einops/blob/master/tests/test_ops.py#L247-L250

Reply:
I just tried this, but it seems there aren't any available symbolic frameworks available for einsum. MXNet seems to not have einsum available for either their symbolic or non-symbolic backends. They have one listed under `mxnet.np.einsum`, but it uses a different internal backend incompatible with `mxnet.nd`. The other symbolic one is keras, but only with `layers=True`, which can't seem to be passed through the functional version of einsum.

Reply:
Actually it seems I might be able to get the Keras one working. But the symbols have one extra leading axis than expected. I guess for those, I need to vectorize the `tf.einsum` over its inputs by default?

Reply:
Okay got a symbolic test working for keras!